kani-compiler/src/kani_queries/mod.rs

```diff
@@ -41,4 +44,40 @@ impl QueryDb {
     pub fn args(&self) -> &Arguments {
         self.args.as_ref().expect("Arguments have not been initialized")
     }
+
+    pub fn add_modifies_contract(&mut self, name: InternedString, contract: AssignsContract) {
+        assert!(
+            self.modifies_contracts.insert(name, contract).is_none(),
+            "Invariant broken, tried adding second modifies contracts to: {name}",
+        )
+    }
+
+    pub fn get_modifies_contracts(&mut self, key: &str) -> Option<AssignsContract> {
+        self.modifies_contracts.remove(&key.intern())
```

Comment:
Why are you removing the contract here?

nit: The rust naming convention is to omit the "get_" prefix. Also, shouldn't this be singular. `modifies_contract`.